### ISSUE TITLE
Problems with straightening transform of Curved Planar Reformat module

### ISSUE THREAD

**Issue by mauigna06** [SlicerIGT/SlicerBoneReconstructionPlanner#50]
Hi @lassoan. 

Without custom plates to ensure non-intersecting holes (of fixing screws and dental implants) in the mandible reconstruction the fibula dental implant drill guides are not so useful because eyeballing of the surgeon becomes necessary. So getting to make custom plates is very important. That way will reach to get the standard of care with BoneReconstructionPlanner.

I'm testing the principle to create the custom plate. I tried to make create the straightening transform of the curved plate using a spline curve (that resambles a linear curve for more precision because the mandible reconstruction is made of linear bone segments and the spline curve will resemble a linear curve when we use it to create a custom plate with this method).

![Screenshot](https://user-images.githubusercontent.com/19158307/124368055-71770380-dc33-11eb-9c48-cf25ed345be1.png)

![Screenshot2](https://user-images.githubusercontent.com/19158307/124368059-763bb780-dc33-11eb-961c-a1418e224566.png)

As you can see I couldn't achieve straightening without deformation of the mesh. There are big deformations (as in the picture) and small deformations (also slightly visible in the picture) that alter the width of the plate making it non-uniform.
I tried with smaller slice size and bigger curve resolution without better results. I don't see folding in the earlier pictures but there are abrupt changes of direction on the grid lines that I assume could have something to do with the bad results.

Even trying with a different curve that was smooth I couldn't achieve results as shown in the [documentation](https://github.com/PerkLab/SlicerSandbox):
![goodResult](https://camo.githubusercontent.com/5cbadcc6659bcaa924043a9177f628acdfdd9560886a857440bc1b80d328b283/68747470733a2f2f617773312e646973636f757273652d63646e2e636f6d2f7374616e6461726431372f75706c6f6164732f736c696365722f6f7074696d697a65642f32582f302f303435303438363936643638336666666166626561383031656465623035636331333439616264355f325f31303335783633302e706e67)

This was the best I got with the smooth curve and there are abrupt changes of direction on the grid lines:
![Screenshot3](https://user-images.githubusercontent.com/19158307/124368146-b0f21f80-dc34-11eb-806e-f131738bb7fa.png)

Do have any idea/suggestions of how to improve this results?

**Comment by mauigna06** [comment]
Things we talked about in the developer meeting:
- Simulating physical plate bending on the computer with previously proposed non-linear transform won't be useful because you cannot guarantee that the surgeon will bend the plate in the simulated way (or that the simulation is accurate to represent tool-bending of the plate).
- 3D printed custom plates for fixing the reconstruction are the standard of care and will focus on that.

- We discussed about 4 ways of making a custom plate mesh:
1. Make it with custom linear extrusion algorithm with the profile curve and the path curve as inputs.
1. Make it from a straight plate mesh (vtkBoxSource) and bend it with a curved planar reformat module's non-linear transform.
1. Make it with the segment editor (grow with margin and cut with scissors) although the mandible reconstruction model will have to be imported to the mandible segmentation so this may not be accurate enough.
1. Make it creating two tubes from one above, one below mandible-reconstruction-curves and join them with small boxes (vtkBoxSource) and make the boolean union of all that and then the boolean difference with the mandible reconstruction model.

- If more accuracy is needed FEA should be used to develop the mesh

Sorry if I forgot something, we talked a lot in the meeting.

@lassoan will check a scene I will upload to see if he can make the non-linear transform straighten the plate without deformation.

@lassoan next steps for me or something you'd like to get me working on?

Custom-plate design is the only thing we lack to achieve the standard of care on mandible reconstruction inside Slicer.
